packages/glimmer-runtime/lib/component/interfaces.ts

```diff
@@ -75,11 +75,20 @@ export interface ComponentAttrsBuilder {
   dynamic(name: string, value: FunctionExpression<string>);
 }
 
+
+const COMPONENT_DEFINITION_BRAND = 'COMPONENT DEFINITION [id=e59c754e-61eb-4392-8c4a-2c0ac72bfcd4]';
+
+export function isComponentDefinition(obj: any): obj is ComponentDefinition<Opaque> {
```

Comment:
please use `typeof obj === 'object' && obj && obj[COMPONENT_DEFINITION_BRAND]` so it doesn't compile a coercion to boolean for any value.

This is common source deopts if obj has a range of types and optimizes the coercion before seeing them all whereas typeof is generic and reduces the coercion to just checking for null.